civiclens/nlp/topics.py

```diff
@@ -222,7 +213,87 @@ def label_topics(topics: dict[int, list], model: TopicChain) -> dict[int, str]:
         Dictionary of topics, and labels
     """
     labels = {}
-    for topic, terms in topics.values():
+    for topic, terms in topics.items():
         labels[topic] = model.generate_label(terms)
 
     return labels
+
+
+def topic_comment_analysis(
+    comment_data: RepComments,
+    model: TopicModel = None,
+    labeler: LabelChain = None,
+    sentiment_analyzer: Callable = None,
+) -> RepComments:
+    """
+    Run topic and sentiment analysis.
+    """
+    comments: list[Comment] = []
+
+    if comment_data.summary:
+        comments += [
+            Comment(text=comment_data.summary, id="Summary", source="Summary")
+        ]
+    try:
+        comments += comment_data.to_list()
+        comment_topics = model.run_model(comments)
+    except TooFewTopics:
+        comments += comment_data.get_nonrepresentative_comments()
+        comment_topics = model.run_model(comments)
+    except Exception:
```

Comment:
want to have a specific exception here if possible and probably a print statement so we can incorporate it into logs at some point